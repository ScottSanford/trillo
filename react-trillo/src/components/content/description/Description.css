.description {
    background-color: var(--color-white);
    box-shadow: var(--shadow-light-1);
    flex: 0 0 60%;
    margin-right: 4.5rem;
    padding: 3rem;

}

@media only screen and (max-width: 56.25em) {
    .description {
        padding: 2rem;
        margin-right: 3rem;
    }
}

@media only screen and (max-width: 37.5em) {
    .description {
        margin-right: 0;
        margin-bottom: 3rem;
    }
}
