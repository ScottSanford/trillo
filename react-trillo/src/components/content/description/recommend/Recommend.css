.recommend {
    align-items: center;
    font-size: 1.3rem;
    color: var(--color-grey-dark-3);
    display: flex;
}

.recommend__count {
    margin-right: auto;
}

.recommend__friends {
    display: flex;
}

.recommend__photo {
    border: 3px solid var(--color-white);
    border-radius: 50%;
    box-sizing: content-box;
    height: 4rem;
    width: 4rem;
}   

.recommend__photo:not(:last-child) {
    margin-right: -2rem;
}
