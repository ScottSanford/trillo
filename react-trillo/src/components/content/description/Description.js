import React from 'react'
import './Description.css'
import Recommend from './recommend/Recommend'
import BenefitsList from './benefits-list/Benefits-List'

export default function Description({ benefits, recommended }) {
	return (
		<div className="description">
        <p className="paragraph">
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Animi nisi dignissimos debitis ratione sapiente saepe. Accusantium
          cumque, quas, ut corporis incidunt deserunt quae architecto voluptate.
            </p>
        <p className="paragraph">
          Accusantium cumque, quas, ut corporis incidunt deserunt quae architecto voluptate delectus, inventore iure aliquid aliquam.
        </p>
        <BenefitsList benefits={benefits} />
        <Recommend list={recommended} />
      </div>
	)
}
