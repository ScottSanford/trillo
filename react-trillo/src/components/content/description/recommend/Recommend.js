import React from 'react'
import './Recommend.css'

const friendRecommendList = [
  { img: 'img/user-3.jpg', name: 'Friend 1'},
  { img: 'img/user-4.jpg', name: 'Friend 2'},
  { img: 'img/user-5.jpg', name: 'Friend 3'},
  { img: 'img/user-6.jpg', name: 'Friend 4'},
]

export default function Recommend() {

  const recommendFriendImages = friendRecommendList.map(friend => {
    return (
      <img key={friend.name} src={friend.img} alt={friend.name} className="recommend__photo" />
    )
  })

  return (
    <div className="recommend">
      <p className="recommend__count">Lucy and 3 other friends recommend this hotel.</p>
      <div className="recommend__friends">
        {recommendFriendImages}
      </div>
    </div>
  )
}