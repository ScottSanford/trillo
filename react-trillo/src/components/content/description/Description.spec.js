import React from 'react'
import { shallow } from 'enzyme'
import Description from './Description'
import BenefitsList from './benefits-list/Benefits-List'
import Recommend from './recommend/Recommend'

describe('Description Component', () => {

	it('should render correctly with no props', () => {

		const wrapper = shallow(<Description />)
		expect(wrapper).toMatchSnapshot()
	})

	it('shallow wrapper instance should be null', () => {
		const wrapper = shallow(<Description />)
		const instance = wrapper.instance()
	  
		expect(instance).toEqual(null)
	  })

	it('should contain 2 paragraph elements', () => {

		const wrapper = shallow(<Description />)
		expect(wrapper.find('.paragraph')).toHaveLength(2)
	})

	it('should render 1 BenefitsList and 1 Recommend list component', () => {

		const wrapper = shallow(<Description />)
		expect(wrapper.find(<BenefitsList />)).toBeTruthy()
		expect(wrapper.find(<Recommend />)).toBeTruthy()
	})
})