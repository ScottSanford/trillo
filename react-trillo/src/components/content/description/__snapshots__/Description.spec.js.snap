// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description Component should render correctly with no props 1`] = `
<div
  className="description"
>
  <p
    className="paragraph"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Animi nisi dignissimos debitis ratione sapiente saepe. Accusantium cumque, quas, ut corporis incidunt deserunt quae architecto voluptate.
  </p>
  <p
    className="paragraph"
  >
    Accusantium cumque, quas, ut corporis incidunt deserunt quae architecto voluptate delectus, inventore iure aliquid aliquam.
  </p>
  <BenefitsList />
  <Recommend />
</div>
`;
